pallets/allocations/src/lib.rs

```diff
@@ -180,88 +207,35 @@ pub mod pallet {
 	#[pallet::generate_store(pub(super) trait Store)]
 	pub struct Pallet<T>(PhantomData<T>);
 
-	#[pallet::hooks]
-	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
-		fn on_initialize(_: BlockNumberFor<T>) -> Weight {
-			let n = Self::relative_block_number();
-			let forced = <NextSessionQuota<T>>::get().is_none();
-			Self::checked_calc_session_quota(n, forced)
-				.saturating_add(Self::checked_renew_session_quota(n, forced))
-				// Storage: Allocations NextSessionQuota (r:1 w:0)
-				// Storage: Allocations MintCurveStartingBlock (r:1 w:0)
-				.saturating_add(T::DbWeight::get().reads(2 as Weight))
-		}
-	}
-
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Optimized allocation call, which will batch allocations of various amounts
 		/// and destinations and together. This allow us to be much more efficient and thus
 		/// increase our chain's capacity in handling these transactions.
-		#[pallet::weight(<T as pallet::Config>::WeightInfo::batch(batch.len().try_into().unwrap_or_else(|_| T::MaxAllocs::get())))]
+		#[pallet::weight(T::WeightInfo::allocate(rewards.len().try_into().unwrap_or_else(|_| T::MaxAllocs::get())).saturating_add(T::WeightInfo::checked_update_session_quota()))]
 		pub fn batch(
 			origin: OriginFor<T>,
-			batch: BoundedVec<(T::AccountId, BalanceOf<T>), T::MaxAllocs>,
+			rewards: BoundedVec<(T::AccountId, BalanceOf<T>), T::MaxAllocs>,
```

Comment:
It is a breaking change for to change this parameter name. It will require a coordinated deployment on the testnet